src/f.jl

```diff
@@ -0,0 +1,107 @@
+# f.jl
+# F-tests
+#
+# Copyright (C) 2019   Raphael Saavedra
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export VarianceFTest
+
+struct VarianceFTest <: HypothesisTest
+    n_x::Int   # size of sample 1
+    n_y::Int   # size of sample 2
+    df_x::Int  # degrees of freedom 1
+    df_y::Int  # degrees of freedom 2
+    F::Real    # test statistic
+    F0::Real   # variance ratio under h_0
+end
+
+"""
+    VarianceFTest(y1::AbstractVector{<: Real}, y2::AbstractVector{<: Real})
+
+Perform an F-test of the null hypothesis that two real-valued vectors `y1` and `y2` have equal variances.
+
+Implements: [`pvalue`](@ref)
+
+# References
+
+  * George E. P. Box, "Non-Normality and Tests on Variances", Biometrika 40 (3/4): 318–335, 1953.
+
+# External links
+
+  * [F-test of equality of variances on Wikipedia](https://en.wikipedia.org/wiki/F-test_of_equality_of_variances)
+"""
+function VarianceFTest(y1::AbstractVector{<: Real}, y2::AbstractVector{<: Real}; F0::Real = 1.0)
+    n1, n2 = length(y1), length(y2)
+    F = var(y1) / var(y2)
+    return VarianceFTest(n1, n2, n1-1, n2-1, F, F0)
+end
+
+"""
+    VarianceFTest(y::AbstractVector{<: Real}, firstlast::Integer = div(length(y), 2))
+
+Perform an homoscedasticity F-test of the null hypothesis that a real-valued vector `y` has constant variance. 
+The test is conducted over the ratio of the sample variances of the first and the last `firstlast` observations 
+of the series.
+
+Implements: [`pvalue`](@ref)
+
+# References
+
+  * James Durbin and Siem Jan Koopman, "Time Series Analysis by State Space Methods", 2nd Ed.,
+    2012, Oxford Statistical Science Series.
+
+# External links
+
+  * [Homoscedasticity on Wikipedia](https://en.wikipedia.org/wiki/Homoscedasticity)
+"""
+function VarianceFTest(y::AbstractVector{<: Real}; firstlast::Integer = div(length(y), 2))
```

Comment:
This isn't correct as it overwrites the previous method (keyword arguments don't participate in dispatch). So you need to use a single method with `firstlast=nothing` by default and handle this internally.